test/extensions/filters/http/decompressor/decompressor_filter_test.cc

```diff
@@ -299,6 +327,20 @@ TEST_P(DecompressorFilterTest, ExplicitlyEnableAdvertiseAcceptEncoding) {
                       "br,mock" /* expected_accept_encoding */);
 }
 
+TEST_P(DecompressorFilterTest, ExplicitlyEnableAdvertiseAcceptEncodingOnlyOnce) {
```

Comment:
This refactor is great -- this is much easier to read and understand the intent of each test. But can you make this be 4 separate test-methods again, each can be this one call? As it is, if there's a failure it will not be visible from logs which of these sub-tests failed.